File: app/src/main/java/com/test/rxjava/fragment/Sample5Fragment.java
package com.test.rxjava.fragment;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;

import com.test.rxjava.utils.LiveDataBus;

public class Sample5Fragment extends Fragment {


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        LiveDataBus.get()
                .with("key_test", String.class)
                .observe(getViewLifecycleOwner(), new Observer<String>() {
                    @Override
                    public void onChanged(@Nullable String s) {
                        Log.i("key_test:onChanged", s);
                    }
                });


        LiveDataBus.get().with("key_test").setValue("111");
    }
}

File: app/src/main/java/com/test/rxjava/fragment/Sample5Fragment.kt
package com.test.rxjava.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.work.Data
import androidx.work.OneTimeWorkRequest
import androidx.work.PeriodicWorkRequest
import androidx.work.WorkManager
import com.test.rxjava.R
import com.test.rxjava.service.Sample3Worker
import com.test.rxjava.service.SampleWork
import java.util.concurrent.TimeUnit

/**
 * work
 */
class Sample5Fragment : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_sample5, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val data = Data.Builder().putString("key","token").build()

        val workManager = context?.let { WorkManager.getInstance(it) }
        workManager?.enqueue(OneTimeWorkRequest.Builder(Sample3Worker::class.java).setInputData(data).build())
//        workManager?.enqueue(PeriodicWorkRequest.Builder(SampleWork::class.java, 20, TimeUnit.MINUTES).build())
    }
}